File: courier_application/app/src/main/java/by/bstu/vs/stpms/courier_application/model/database/dao/OrderDao.java
package by.bstu.vs.stpms.courier_application.model.database.dao;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import java.util.List;

import by.bstu.vs.stpms.courier_application.model.database.entity.Order;

@Dao
public abstract class OrderDao extends AbstractDao<Order>  {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    public abstract void insertAll(List<Order> orders);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    public abstract void insertWithReplace(Order order);

    @Override
    @Query("SELECT * FROM orders WHERE id = :id")
    public abstract LiveData<Order> findById(long id);

    @Override
    @Query("SELECT * FROM orders")
    public abstract LiveData<List<Order>> getAll();
}

File: courier_application/app/src/main/java/by/bstu/vs/stpms/courier_application/model/database/dao/OrderDao.kt
package by.bstu.vs.stpms.courier_application.model.database.dao

import androidx.room.*
import by.bstu.vs.stpms.courier_application.model.database.entity.*

@Dao
abstract class OrderDao : AbstractDao<Order>() {

    @Query("DELETE FROM orders WHERE 1=1")
    abstract override suspend fun truncate()

    @Query("SELECT * FROM orders WHERE id = :id")
    abstract override suspend fun findById(id: Long): Order?

    @Query("SELECT * FROM orders")
    abstract override suspend fun getAll(): List<Order>
}